dashboard/export_scores.R

```diff
@@ -0,0 +1,42 @@
+source('./common.R')
+
+create_export_df = function(scoreDf, targetVariable, forecasters, horizon, loc) {
+    signalFilter = CASE_FILTER
+    if (targetVariable == "Deaths") {
+      signalFilter = DEATH_FILTER
+    }
+    scoreDf = scoreDf %>%
+      filter(signal == signalFilter) %>%
+      filter(ahead %in% horizon) %>%
+      filter(forecaster %in% forecasters)
+    if (loc != TOTAL_LOCATIONS) {
```

Comment:
It seems that if TOTAL_LOCATIONS is true, this download actually includes all locations. The downloaded file (with just the initial selections on app load selected) included AS, although it technically shouldn't since the baseline nor ensemble forecasters have data for that.

Reply:
There is probably a cleaner way of doing it, but this is what I did in the app to find and reduce by the locations that all forecasters have data for:

```# Create df with col for all locations across each unique date, ahead and forecaster combo
      locationDf = filteredScoreDf %>% group_by(Forecaster, Week_End_Date, ahead) %>%
        summarize(location_list = paste(sort(unique(geo_value)),collapse=","))

      # Create a list containing each row's location list
      locationList = sapply(locationDf$location_list, function(x) strsplit(x, ","))
      locationList = lapply(locationList, function(x) x[x != 'us'])

      # Get the intersection of all the locations in these lists
      locationsIntersect = unique(Reduce(intersect, locationList))
      filteredScoreDf = filteredScoreDf %>% filter(geo_value %in% locationsIntersect)```

Reply:
Also now that I am adding hospitalizations which require even more data preprocessing (because of the difference in how the aheads work) this export function will probably need even more adjustments and require more decisions of what we actually want to show (the raw data vs the processed/filtered data that is shown in the plot). I can bring this up in our next meeting to get thoughts on what we want users to download.